cupy/_core/_gufuncs.py

```diff
@@ -191,15 +192,15 @@ def _process_signatures(self, signatures):
             ops.append(_OpsRegister._Op(ins, outs, op))
         return ops
 
-    def _determine_from_args(self, args, casting):
+    def _determine_from_args(self, args):
         n = len(args)
         in_types = tuple(arg.dtype for arg in args)
         for op in self._ops:
             op_types = op.in_types
             for i in range(n):
                 it = in_types[i]
                 ot = op_types[i]
-                if not numpy.can_cast(it, ot, casting=casting):
+                if not numpy.can_cast(it, ot):
```

Comment:
CI failure of TestGUFuncSignatures seems to be related to this fix. This code is certainly unused from matmul, but we want to keep the support of `signatures` option.

Reply:
Arg, sorry for not realizing... I guess I can just undo the change.  There may also be a point in using `casting="safe" if casting in ["unsafe", "same_kind"] else casting` to allow fallback to default when `casting="no"` is given.  In principle, I think casting shouldn't matter, but with a fallback to `default`, I guess stricter casting does matter.

(not sure it is worth trying to figure out compared to just undoing...)

Reply:
Sorry for not doing it earlier.  I just pulled it out for now.  I think the above is more correct (currently `casting="unsafe"` would just pick the first loop, which is certainly correct).

But, there was always too much in here, give me a nudge and I will make sure to follow up.